curator/utils.py

```diff
@@ -745,36 +751,48 @@ def get_client(**kwargs):
                     import certifi
                     kwargs['verify_certs'] = True
                     kwargs['ca_certs'] = certifi.where()
-
     try:
         from requests_aws4auth import AWS4Auth
-        kwargs['aws_key'] = False if not 'aws_key' in kwargs \
-            else kwargs['aws_key']
-        kwargs['aws_secret_key'] = False if not 'aws_secret_key' in kwargs \
-            else kwargs['aws_secret_key']
-        kwargs['aws_region'] = False if not 'aws_region' in kwargs \
-            else kwargs['aws_region']
+        kwargs['aws_sign_request'] = False if not 'aws_sign_request' in kwargs \
+            else kwargs['aws_sign_request']
+        if kwargs['aws_sign_request']:
```

Comment:
This is going to need to be more complex.

Not only does it have to `try`/`except` on the import module, it has to `try`/`except` on the `session.Session()` (because it will raise an exception if it is unable, and we should control how it acts if it does, and log good messages).  If it is unable to get the session, then it would then rely on the original `aws_key` and `aws_secret_key`, which should be set first.

The contents of the `else` block should precede the `if kwargs['aws_sign_request']` statement as a pre-population thing, even if they end up being `False` because they were not specified.  The `boto3` stuff will overwrite the necessary values if it exists, so it is completely okay to have `kwargs['aws_key']` and `kwargs['aws_secret_key']` set first.

Does this make sense?

Reply:
This makes sense. I have implemented your suggestions, so now credentials retrieved with boto3 will overwrite anything that was in the config file if the user specified both.